### PR TITLE
[Bugfix][Test] Fix off-by-one error in sampled token rank causing flaky logprobs test

### PR DESCRIPTION
**Pull request by mayuyuace**
## Summary

Fixes a non-deterministic (flaky) test failure in:
  tests/v1/engine/test_output_processor.py::test_logprobs_processor[None-5-RequestOutputKind.DELTA]

https://buildkite.com/vllm/intel-ci/builds/9778/canvas?sid=01a03c1d-1b9a-43d7-8c99-b01d13ceabdc&tab=output

## Root Cause

In `tests/v1/engine/utils.py`, `_create_random_top_token_test_vector()` builds a
test fixture that includes a `sampled_token_rank` — the expected rank of the sampled
token in the logprobs output.

`torch.randperm` is used to generate `choice_tensor`, and `choice_tensor[0]` is then
forcibly set to `sampled_token_id`. If `randperm` happens to also place
`sampled_token_id` somewhere in `choice_tensor[1:]` (Case A), the function computes
the rank as:
```
# Before (buggy)
sampled_token_rank = (
    (choice_tensor[1:] == sampled_token_id).nonzero(as_tuple=True)[0].item()
)
```
This returns a 0-based index into choice_tensor[1:].
However, vllm/logprobs.py::append_logprobs_for_next_position assigns top-k ranks
using range(1, num_logprobs + 1) — 1-based. When sampled_token_id appears
twice in token_ids (at index 0 and somewhere in the top-k), the dict entry gets
overwritten by the 1-based top-k rank, while ref_sampled_token_rank holds the
0-based value. The assertion at test line 293 then fails:
assert ref_sampled_token_rank == smp_lp_rank
**Why It's Flaky**
The bug only triggers when torch.randperm places sampled_token_id in
choice_tensor[1:]. The probability per token position is approximately
num_logprobs / vocab_size (e.g. 5 / 128000 ≈ 0.004%). The torch random seed
is never fixed in utils.py, so the outcome varies across runs. Rare enough to
pass most of the time, but consistently observed across CI runs over time.
**Fix**
Add + 1 to convert the 0-based index to a 1-based rank, matching the convention
used by append_logprobs_for_next_position:
```
# After (fixed)
sampled_token_rank = (
    (choice_tensor[1:] == sampled_token_id).nonzero(as_tuple=True)[0].item()
    + 1
)
```

## Testing

The fix is in test utility code only (`tests/`), no production code is changed.

A standalone reproducer that deterministically constructs the worst-case
`choice_tensor` and confirms the bug (and the fix) can be run with:

```python
# Requires: torch
# Demonstrates the off-by-one without needing a lucky torch.randperm

import itertools, sys, torch

NUM_LOGPROBS = 5
sampled_token_id = 3021

# Simulate randperm placing sampled_token_id in both [0] and [3]
choice_tensor = torch.tensor([sampled_token_id, 100, 200, sampled_token_id, 400, 500])

# --- Buggy behavior (before fix) ---
ref_rank_buggy = (
    (choice_tensor[1:] == sampled_token_id).nonzero(as_tuple=True)[0].item()
    # returns 2 (0-based), but consumer expects 1-based
)

logprob_vals = [-10.0, -1.0, -2.0, -3.0, -4.0, -5.0]
topk_ranks = range(1, NUM_LOGPROBS + 1)
ranks = list(itertools.chain((ref_rank_buggy,), topk_ranks))

result_dict = {}
for token_id, lp, rank in zip(choice_tensor.tolist(), logprob_vals, ranks):
    result_dict[token_id] = rank

smp_lp_rank = result_dict[sampled_token_id]
assert ref_rank_buggy != smp_lp_rank  # confirms bug: 2 != 3

# --- Fixed behavior (after fix) ---
ref_rank_fixed = (
    (choice_tensor[1:] == sampled_token_id).nonzero(as_tuple=True)[0].item() + 1
)
ranks_fixed = list(itertools.chain((ref_rank_fixed,), topk_ranks))

result_dict_fixed = {}
for token_id, lp, rank in zip(choice_tensor.tolist(), logprob_vals, ranks_fixed):
    result_dict_fixed[token_id] = rank

smp_lp_rank_fixed = result_dict_fixed[sampled_token_id]
assert ref_rank_fixed == smp_lp_rank_fixed  # fix works: 3 == 3
print("OK")

### CODE REVIEW

**Review by njhill — COMMENTED** [review]
Thanks @mayuyuace, could you also make the equivalent fix to the adjacent `_create_random_top_token_test_matrix` method?

**Comment by mayuyuace** [comment]
> _create_random_top_token_test_matrix

Done, thanks for catching that! Applied the same + 1 fix to _create_random_top_token_test_matrix as well.

**Review by njhill — APPROVED** [review]
